### DISCUSSION TITLE
Changing fragment file locations for multiple samples in the same object

### DISCUSSION THREAD

**Discussion by Connorr0** [stuart-lab/signac#662] [Q&A] [answered]
Hey, I have some integrated objects that I am hoping to do CoveragePlots on. All of our objects are stored on the server and each computer accesses the server through a different path (mac vs pc). I know there is a way to change fragment paths for single samples, but is there a way to do that for multiple samples?

**Comment by timoast** [comment, marked as answer]
You will need to run `UpdatePath()` on each fragment object.

**Reply by Connorr0** [reply]
For clarification I tried this:

```
for (f in 1:length(Fragments(d87all_int_DS))){
  print(f)
  frags_obj <- Fragments(d87all_int_DS)[[f]]
  
  frag_path = GetFragmentData(Fragments(d87all_int_DS)[[f]])
  
  frag_path = stringr::str_replace(frag_path, "/Volumes/Data", replacement = "X:")
  
  Fragments(d87all_int_DS)[[f]] <- UpdatePath(frags_obj, frag_path)
}
```

But got the following error:
```
Computing hash
Error in AddFragments(object = object, fragments = value[[i]]) : 
  Cells already present in a fragment object
```

Should I instead be building a new list of objects and replacing them all at the end?

**Reply by Connorr0** [reply]
This worked, but I wasn't sure if there are checks in place that make sure the fragments are in the right order:

```
frag_lst_new = list()
for (f in 1:length(Fragments(d87all_int_DS))){
  print(f)
  frags_obj <- Fragments(d87all_int_DS)[[f]] # assuming there's only one fragment object in the assay
  
  frag_path = GetFragmentData(Fragments(d87all_int_DS)[[f]])
  
  frag_path = stringr::str_replace(frag_path, "/Volumes/Data", replacement = "X:")
  
  frag_lst_new[[f]] <- UpdatePath(frags_obj, frag_path)
}


Fragments(d87all_int_DS) <- NULL
Fragments(d87all_int_DS) <- frag_lst_new
```

**Reply by Connorr0** [reply]
Also, will the second approach work after cells have been renamed to have unique names post-merge/integration?

**Reply by timoast** [reply]
`UpdatePath()` will change the path and check that the same fragment file is present at the new path by comparing the md5 hash. It does not change the cell mapping information stored in the Fragment object (the dictionary of cell names in the Seurat object to cell names in the fragment file).

> This worked, but I wasn't sure if there are checks in place that make sure the fragments are in the right order

Not sure what you mean exactly...the order of fragment objects in the list doesn't matter
